src/store/registry.js

```diff
@@ -10,8 +10,18 @@ export class StoreRegistry {
     this.defaultScope = defaultScope;
   }
 
-  initStore = (key, Store) => {
+  initStore = (key, Store, fromContainer) => {
     const { initialState, actions } = Store;
+
+    if (Store.containedBy && !fromContainer) {
```

Comment:
So it's not checking that container is the one indicated by `containedBy`, allowing any other container to be present.
Question - how I can disable or satisfy this condition in tests, where isolation might be not as required, or where I don't want to fix too many things at once?

Reply:
> allowing any other container to be present

Only the ones targeting the same type! The store creation is triggered by a matching container, not any container.

>  how I can disable or satisfy this condition in tests

There are 2 patterns devs can use in tests:
- use the `containedBy` container directly
- use a new `createContainer(Store)`, which also allows to override `onInit/onUpdate`

The latter might be better for storybooks/tests, but as long as there is a container that captures the store state somewhere in the tree, it will work without complaining. 
Should also help pushing teams writing tests without accidentally leaking to global when `containedBy` is used.